src/lsm/table.zig

```diff
@@ -239,52 +243,22 @@ pub fn TableType(
         pub const block_count_max =
             index_block_count + filter_block_count_max + data_block_count_max;
 
-        const index = struct {
-            const size = @sizeOf(vsr.Header) + filter_checksums_size + data_checksums_size +
-                keys_size + filter_addresses_size + data_addresses_size;
-
-            const filter_checksums_offset = @sizeOf(vsr.Header);
-            const filter_checksums_size = filter_block_count_max * checksum_size;
-
-            const data_checksums_offset = filter_checksums_offset + filter_checksums_size;
-            const data_checksums_size = data_block_count_max * checksum_size;
-
-            const keys_offset = data_checksums_offset + data_checksums_size;
-            const keys_size = data_block_count_max * key_size;
-
-            const filter_addresses_offset = keys_offset + keys_size;
-            const filter_addresses_size = filter_block_count_max * address_size;
-
-            const data_addresses_offset = filter_addresses_offset + filter_addresses_size;
-            const data_addresses_size = data_block_count_max * address_size;
-
-            const padding_offset = data_addresses_offset + data_addresses_size;
-            const padding_size = block_size - padding_offset;
-        };
-
-        pub const filter = struct {
-            pub const data_block_count_max = layout.filter_data_block_count_max;
-
-            const filter_offset = @sizeOf(vsr.Header);
-            const filter_size = block_size - filter_offset;
-
-            const padding_offset = filter_offset + filter_size;
-            const padding_size = block_size - padding_offset;
-        };
-
-        pub const data = struct {
-            const key_count = layout.block_key_count;
-            pub const block_value_count_max = layout.block_value_count_max;
-
-            const key_layout_offset = @sizeOf(vsr.Header);
-            const key_layout_size = layout.block_key_layout_size;
+        const index = TableIndexSchema.init(.{
+            .key_size = key_size,
+            .filter_block_count_max = filter_block_count_max,
+            .data_block_count_max = data_block_count_max,
+        });
 
-            const values_offset = key_layout_offset + key_layout_size;
-            const values_size = block_value_count_max * value_size;
+        const filter = TableFilterSchema.init(.{
+            .data_block_count_max = layout.filter_data_block_count_max,
+        });
 
-            const padding_offset = values_offset + values_size;
-            const padding_size = block_size - padding_offset;
-        };
+        pub const data = TableDataSchema.init(.{
+            .key_count = layout.block_key_count,
+            .key_layout_size = layout.block_key_layout_size,
+            .value_count_max = layout.block_value_count_max,
+            .value_size = value_size,
+        });
```

Comment:
These are the schemas which encode _new_ blocks, so they are still comptime-known.